relay_test.go

```diff
@@ -369,7 +369,7 @@ func TestTimeoutCallsThenClose(t *testing.T) {
 func TestLargeTimeoutsAreClamped(t *testing.T) {
 	// TODO: enable frame pool checks
 	const (
-		clampTTL = time.Millisecond
+		clampTTL = 10 * time.Millisecond
 		longTTL  = time.Minute
```

Comment:
(Another way to think about this test)

Currently, it's written to ensure that `longTTL` isn't used, and `clampTTL` is used (and tries to validate that the timeout happens roughly after `clampTTL`. As you called out in the summary, noise can cause the actual time taken to be significantly higher. I wouldn't be surprised to see O(100ms) in some extreme cases. For the test to run as fast as possible however, we want `clampTTL` to be a low as possible, since it's the minimum time taken for the test.

What we really care about is that `longTTL` is clamped down. We don't necessarily need to validate that it's clamped to `clampTTL` (since it's noisy), just the fact that it's clamped is a useful signal that clamping is working.

To make the test less flaky, I think we can:
 * Rely on the existing check in the server-side echo func to verify the actual timeout is less than `clampTTL` (that should never be flaky)
 * Verify that the time passed till timeout is << `longTTL`. E.g., that time taken is less than `longTTL / 2`.

That allows `clampTTL` to stay low, verifies the rewritten timeout (via the server func), and ensures that the actual timeout used on the relay is clamped down. And at the same time, we keep `clampTTL` low, and likely make the test even less flaky than the current proposed change.